realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -76,30 +76,53 @@
     private RealmCache realmCache;
     protected SharedRealm sharedRealm;
     private boolean shouldCloseSharedRealm;
+    private SharedRealm.SchemaChangedCallback schemaChangedCallback = new SharedRealm.SchemaChangedCallback() {
+        @Override
+        public void onSchemaChanged() {
+            RealmSchema schema = getSchema();
+            if (schema != null) {
+                schema.refresh();
```

Comment:
maybe change the name of `RealmSchema.refresh()` to `RealmSchema.invalidate()`? doesn't that sound better?